### ISSUE TITLE
6.2.0 does not redirect to login when accessing protected route

### ISSUE THREAD

**Issue by havard024** [ElderByte-/ngx-jwt-auth#30]
Hi,

when I access a protected route using the SimpleAuthGuard example, nothing happens. 
It seems like the observable for `checkAndHandleIfNotAuthorized` is not resolved.

I managed to get it to work by setting an initial value for the `_principalSubject` in `PrincipalService` like so:

`constructor() {
    this.setPrincipalInternal(null)
 }`

though that seems rather hacky, my suspicion is that there is no initial auth check so the principal is never set thus never resolved.

Library version:
```
├─┬ @elderbyte/ngx-jwt-auth@6.2.0
├── @elderbyte/ngx-simple-webstorage@6.0.0
├── @elderbyte/ts-logger@0.1.0
```
Angular version:
```
    "@angular/animations": "6.0.0",
    "@angular/common": "6.0.0",
    "@angular/compiler": "6.0.0",
    "@angular/core": "6.0.0",
    "@angular/forms": "6.0.0",
    "@angular/http": "6.0.0",
    "@angular/platform-browser": "6.0.0",
    "@angular/platform-browser-dynamic": "6.0.0",
    "@angular/router": "6.0.0",
```
Do you have any input on this or is there anything I can do to assist further debugging this issue?

**Comment by IsNull** [comment]
Thank you for reporting an issue. 

I have to check if there is an issue with `SimpleAuthGuard`, for the mean time you can use the more powerful `HasRoleGuard` which works properly:

```javascript
           {
                path: 'my-route',
                component: MyComponent,
                canActivate: [HasRoleGuard],
                data: {
                    roles: ['USER']
                }
            }
```

**Comment by IsNull** [comment]
@havard024 The `SimpleAuthGuard` appears to work as designed. Maybe it is a general configuration issue.

Can you share your main module configuration and your route configuration?

**Comment by havard024** [comment]
Thank you for the quick response.

I tried with the `HasRoleGuard` as well, but the observable only resolves if I don't configure any roles like so:
```json
data: {
  "roles": ["USER"]
}
```

The configuration I have is exactly the same is in the demo, though I tried to integrate the library within the [ngx admin template](https://github.com/akveo/ngx-admin)

To rule out any issues with the ngx admin template, I'll try harder to get the demo running locally.

**Comment by IsNull** [comment]
> To rule out any issues with the ngx admin template, I'll try harder to get the demo running locally.

It should be as easy as:

* Check out this repo
* Run `npm i` in the checked out folder
* Run `npm run build` -> build the library locally
* Run `npm run start` -> start the demo app, which will use the just build library

**Comment by havard024** [comment]
Managed to build and run the demo, no issues with routing there, though hit one snag while running `npm run build`.

```sh
Bundling SCSS...
Could not find scss source files in: /Users/mac/github/ngx-jwt-auth ./projects/elderbyte/ngx-jwt-auth/src/_theming.scss
imports: undefined
```

had to create an empty file `_theming.scss` then everything worked.

I tried to downgrade the angular and rxjs versions in this library to the same versions in the ngx-admin library and the demo still works. So then there must be something off with the template which I'm not able to figure out.

I'll close this issue since it's not an issue with your library
